src/rgb/schemata/fungible.rs

```diff
@@ -139,26 +148,26 @@ impl Schemata for Rgb1 {
                     Self::BALANCE_SEAL => Amount,
                     Self::PRUNE_SEAL => NoState
                 },
-                transitions: vec![
+                transitions: map!{
```

Comment:
This was my original code at some point in time, but I moved from the map structure to more straingforward (in serialization) Vecs. I understand that map hives simplier validation code, but my original purpose was to have a DSL-style schema definitions so they may be even written as a standalone file.

I would not insist that this is high priority, but I think it worth preferring readability of the schema over validation code complexity. The reason is the following: validation code is written once, peer-reviewed and checked multiple times; schema will be written many times and most of vendor-specific schemas would not be reviewed. This means we need to give schema vendors tooling to do this not in a error-prone way. We, for sure, can create a language for that (like Ethereum did), but I really think that's a bad idea. So let's just use Rust DLS capabilities <https://doc.rust-lang.org/rust-by-example/macros/dsl.html> to achieve that, avoiding all non-trivial parts (like mixing non-necessary id's)

Reply:
the problem here is basically the same as above with the fields: you need to match a transition type to the schema when you want to validate